src/engine/GroupBy.cpp

```diff
@@ -759,57 +760,162 @@ GroupBy::checkIfHashMapOptimizationPossible(
-bool GroupBy::findAggregateMultiple(
+bool GroupBy::findAggregatesImpl(
     sparqlExpression::SparqlExpression* parent,
     sparqlExpression::SparqlExpression* expr, std::optional<size_t> index,
     std::vector<GroupBy::HashMapAggregateInformation>& info) {
   if (dynamic_cast<sparqlExpression::AvgExpression*>(expr) != nullptr) {
-    info.push_back({expr, parent, index});
+    if (parent != nullptr && index.has_value()) {
+      ParentAndChildIndex parentAndChildIndex{parent, index.value()};
+      info.emplace_back(expr, 0, parentAndChildIndex);
+    } else
+      info.emplace_back(expr, 0, std::nullopt);
     return true;
   }
 
   // Unsupported aggregates
-  if (dynamic_cast<sparqlExpression::SumExpression*>(expr) != nullptr ||
-      dynamic_cast<sparqlExpression::MinExpression*>(expr) != nullptr ||
-      dynamic_cast<sparqlExpression::MaxExpression*>(expr) != nullptr ||
-      dynamic_cast<sparqlExpression::CountExpression*>(expr) != nullptr ||
-      dynamic_cast<sparqlExpression::GroupConcatExpression*>(expr) != nullptr) {
-    return false;
-  }
+  if (!isSupportedAggregate(expr)) return false;
 
   auto children = expr->children();
 
-  std::vector<bool> childrenContainUnsupportedAggregate;
-  // TODO<C++23> use views::enumerate
+  bool childrenContainOnlySupportedAggregates = true;
   size_t childIndex = 0;
   for (const auto& child : children) {
-    childrenContainUnsupportedAggregate.push_back(
-        findAggregateMultiple(expr, child.get(), childIndex++, info));
+    childrenContainOnlySupportedAggregates =
+        childrenContainOnlySupportedAggregates &&
+        findAggregatesImpl(expr, child.get(), childIndex++, info);
   }
 
-  return std::all_of(childrenContainUnsupportedAggregate.begin(),
-                     childrenContainUnsupportedAggregate.end(),
-                     [](const bool v) { return v; });
+  return childrenContainOnlySupportedAggregates;
+}
+
+// _____________________________________________________________________________
+void GroupBy::extractValues(
+    sparqlExpression::ExpressionResult&& expressionResult,
+    sparqlExpression::EvaluationContext& evaluationContext,
+    IdTable* resultTable, LocalVocab* localVocab, size_t outCol) {
+  auto visitor = [&evaluationContext, &resultTable, &localVocab,
+                  &outCol]<sparqlExpression::SingleExpressionResult T>(
+                     T&& singleResult) mutable {
+    auto generator = sparqlExpression::detail::makeGenerator(
+        std::forward<T>(singleResult),
+        evaluationContext._endIndex - evaluationContext._beginIndex,
+        &evaluationContext);
+
+    // TODO<C++23> use views::enumerate
+    auto idx = 0;
+    for (sparqlExpression::IdOrString val : generator) {
+      (*resultTable)(evaluationContext._beginIndex + idx++, outCol) =
+          sparqlExpression::detail::constantExpressionResultToId(std::move(val),
+                                                                 *localVocab);
```

Comment:
Use `auto targetIterator = resultTable->getColumn(outCol).begin() + evaluationContext._beginIndex;`
Then you have the column based interface and inside your loop you can do somethink like
`*targetIterator = constantExpr(....); ++targetIterator`.